repos/system_upgrade/common/actors/peseventsscanner/libraries/peseventsscanner_repomap.py

```diff
@@ -0,0 +1,300 @@
+from leapp.libraries.common.config import get_target_product_channel
+from leapp.libraries.common.config.version import get_source_major_version, get_target_major_version
+from leapp.libraries.stdlib import api
+
+
+DEFAULT_PESID = {
+    '7': 'rhel7-base',
+    '8': 'rhel8-BaseOS',
+    '9': 'rhel9-BaseOS'
+}
+
+
+def _get_channel_prio(pesid_repo):
+    priorities = {
+        'beta': 0,
+        'ga': 1,
+    }
+    return priorities.get(pesid_repo.channel, 10)
+
+
+class RepoMapDataHandler(object):
+    """
+    Provide the basic functionality to work with the repository data easily.
+    """
+
+    def __init__(self, repo_map, default_channels=None, prio_channel=None):
+        """
+        Initialize the object based on the given RepositoriesMapping msg.
+
+        Expects that msg contains just stuff related for the current IPU
+        (at least mapping and repos for the used upgrade path and architecture).
+
+        :param repo_map: A valid RepositoryMapping message.
+        :type repo_map: RepositoryMapping
+        :param default_channels: A list of default channels to use when a target repository
+                                 equivalent exactly matching a source repository was not found.
+        :type default_channels: List[str]
+        :param prio_channel: Prefer repositories with this channel when looking for target equivalents.
+        :type prio_channel: str
+        """
+        # NOTE: currently I am keeping this default data structure that is not
+        # ideal for work, but there is not any significant impact..
+        self.repositories = repo_map.repositories
+        self.mapping = repo_map.mapping
+        # FIXME(pstodulk): what about default_channel -> fallback_channel
+        # hardcoded always as ga? instead of list of channels..
+        # it'd be possibly confusing naming now...
+        self.default_channels = default_channels or ['ga']
+
+        # Make self.prio_channel None if the user did not specify any target channels, so that self.default_channels
+        # will be used instead
+        self.prio_channel = prio_channel or get_target_product_channel(default=None)
```

Comment:
Somehow I don't understand why the `prio_channel` parameter exists, it's not used anywhere and even if there is a reason for it, why it supersedes the thing provided on the command line or environment variable. This seems counter intuitive for me. If we specify the `prio_channel` parameter to be 'ga' anywhere in the code, then a user won't be able to override that anymore. Is that intended?

Reply:
The parameter was present in the original new repomapping implementation, so I let it there even when it was clear that the desired target channel will be taken from environmental vars, thinking it would be cool if the actors that do want to use the library have easy way to override it if the need it.

I really appreciate you raising this concern -  it made me rethink what we are trying achieve in the bigger picture (we want the `src_repoid -> dst_repoid` translation to be consistent across all actors needing it), and this hypothetically useful parameter conflicts it. Therefore, I will remove it. Thanks :+1:.